packages/admin-stories/src/admin/table/TableWithFilterBar.tsx

```diff
@@ -0,0 +1,160 @@
+
+import { apolloStoryDecorator } from "../../apollo-story.decorator";
+
+const gqlRest = gql;
+
+const query = gqlRest`
+query users(
+    $pathFunction: any
+    $username: String
+    $name: String
+    $email: String
+    $website: String
+) {
+    users(
+        email: $email
+        username: $username
+        name: $name
+        website: $website
+    ) @rest(type: "User", pathBuilder: $pathFunction) {
+        id
+        name
+        username
+        email
+        phone
+        website
+    }
+}
+`;
+function pathFunction({ args }: { args: { [key: string]: any } }) {
+    interface IPathMapping {
+        [arg: string]: string;
+    }
+    const paramMapping: IPathMapping = {
+        email: "email",
+        username: "username",
+        name: "name",
+        website: "website",
+    };
+
+    const q = Object.keys(args).reduce((acc: { [key: string]: any }, key: string): { [key: string]: any } => {
+        if (paramMapping[key] && args[key]) {
+            acc[paramMapping[key]] = args[key];
+        }
+        return acc;
+    }, {});
+
+    return `users?${qs.stringify(q, { arrayFormat: "brackets" })}`;
+}
+
+interface IQueryData {
+    users: Array<{
+        id: number;
+        name: string;
+        username: string;
+        email: string;
+        phone: string;
+        website: string;
+    }>;
+}
+
+interface IFilterValues {
+    username: any;
+    name: any;
+    email: any;
+    website: any;
+}
+
+interface IVariables extends IFilterValues {
+    pathFunction: any;
+}
+
+const ExampleWithSelect: React.FC = () => {
+    const options = [
+        { value: "Sincere@april.biz", label: "Sincere@april.biz" },
+        { value: "Shanna@melissa.tv", label: "Shanna@melissa.tv" },
+        { value: "Nathan@yesenia.net", label: "Nathan@yesenia.net" },
+    ];
+    return <Field name="email" type="text" component={FinalFormReactSelectStaticOptions} fullWidth options={options} />;
+};
+
+function Story() {
+    const filterApi = useTableQueryFilter<Partial<IFilterValues>>({});
+
+    const { tableData, api, loading, error } = useTableQuery<IQueryData, Partial<IVariables>>()(query, {
+        variables: {
+            email: filterApi.current.email?.email,
+            name: filterApi.current.name?.name,
+            username: filterApi.current.username?.username,
+            website: filterApi.current.website?.url,
+            pathFunction,
+        },
+        resolveTableData: (data) => ({
+            data: data.users,
+            totalCount: data.users.length,
+        }),
+    });
+
+    return (
+        <TableQuery api={api} loading={loading} error={error}>
+            <TableFilterFinalForm filterApi={filterApi}>
+                <Typography variant="h5">FilterBar</Typography>
+                <FilterBar>
+                    <FilterBar.PopOverFormField label={"Username"} name="username" fieldBarWidth={150}>
+                        <Field name="username" type="text" component={FinalFormInput} fullWidth />
```

Comment:
Why does the name have to be specified twice?

Reply:
because the Field needs a name (Field of the internal Form), and also de Field(field of the TableFIlterFinalForm component) needs a name prop.

Reply:
In each FilterBar.Field is typically a group of other fields. The name-attr of FilterBar.Field represents the group-name, and the name-attr of  Field represents the name of the field.

In cases where only one Field is in a group, the duplication of the name-value seems redundant, but it's ok IMO.

Reply:
my idea how it could be avoided https://github.com/vivid-planet/comet-admin/pull/378

Reply:
I am not sure if this is better: https://github.com/vivid-planet/comet-admin/pull/378#discussion_r634188881